internal/gatewayapi/resource.go

```diff
@@ -48,7 +48,8 @@ type Resources struct {
 	EndpointSlices          []*discoveryv1.EndpointSlice   `json:"endpointSlices,omitempty" yaml:"endpointSlices,omitempty"`
 	Secrets                 []*v1.Secret                   `json:"secrets,omitempty" yaml:"secrets,omitempty"`
 	ConfigMaps              []*v1.ConfigMap                `json:"configMaps,omitempty" yaml:"configMaps,omitempty"`
-	EnvoyProxy              *egv1a1.EnvoyProxy             `json:"envoyProxy,omitempty" yaml:"envoyProxy,omitempty"`
+	ClassEnvoyProxy         *egv1a1.EnvoyProxy             `json:"classEnvoyProxy,omitempty" yaml:"classEnvoyProxy,omitempty"`
+	EnvoyProxies            []*egv1a1.EnvoyProxy           `json:"EnvoyProxies,omitempty" yaml:"EnvoyProxies,omitempty"`
```

Comment:
I know this is needed for different targets, but a decision should be made here:

- do we accept keeping one type of resource two places, they are all envoyproxies, is there any better definition for this?

/cc @envoyproxy/gateway-maintainers

Reply:
which struct making code simpler to read  :) ? thats my vote

Reply:
okay for me, but some comments need to be added, like which one is used by GC, which one is used by GTWs.

Reply:
> but some comments need to be added

added